### PR TITLE
fix(medusa): apply discount with product module

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- include `variant.product_id` as field when populating variants for cart in cart-update

**Why**
- without a `product_id` validating the product for discount always returns false

### CODE REVIEW

**Review by olivermrbl — APPROVED** [review]

**Comment by olivermrbl** [comment]
@pKorsholm – can I get you to try to add:
```
      - name: Build Packages
        run: yarn build
```

to the steps of the `integration-tests-repositories`? After `Install dependencies`

It seems the test suite is flaky, because it fails to find files from a build.

File: integration-tests/plugins/__tests__/cart/store/index.ts
Comment (olivermrbl) [inline integration-tests/plugins/__tests__/cart/store/index.ts]: rm

File: integration-tests/plugins/__tests__/cart/store/index.ts
Comment (olivermrbl) [inline integration-tests/plugins/__tests__/cart/store/index.ts]: rm `.skip`

File: integration-tests/plugins/__tests__/cart/store/index.ts
Comment (olivermrbl) [inline integration-tests/plugins/__tests__/cart/store/index.ts:225]: suggestion: use product factory?

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Comment by olivermrbl** [comment]
@pKorsholm, can you attend to the last comments so we can merge? 🙏

File: packages/medusa/src/api/routes/store/carts/update-cart.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/store/carts/update-cart.ts:147]: question: shouldn't the __args to filter be in __args: { filters: { } } ? cc @carlos-r-l-rodrigues.

**Review by adrien2p — APPROVED** [review]
LGTM, after oli's comments and my question. we need to align on all places where we are using the remote query. looks like sometimes the constratins are in filters and other times they are not. though, this part does not block the pr if it works. Also, maybe one thing to consider in future pr is to use the remote-query utils to have something more human understandable :), __args seems a bit abstract and we don't know what it represents as a user reading the codebase. the utils abstract those as well as simplifying the API. cc @olivermrbl

File: packages/medusa/src/api/routes/store/carts/update-cart.ts
Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/api/routes/store/carts/update-cart.ts:147]: when the property is a primary key (defined in the "__joinerConfig" as a primary key), the `id` in this case, by default it is used as a filter.

Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/api/routes/store/carts/update-cart.ts:147]: Also the `product_id` in this case has to be manually added because it is an internal relation of the Product module.
If it was a relation represented on the joiner configs, it would be auto infered.

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

**Pull request closed without merging**
